proposals/shmsize-tmpfs.md

```diff
@@ -0,0 +1,46 @@
+<!--
+Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"). You may
+not use this file except in compliance with the License. A copy of the
+License is located at
+
+     http://aws.amazon.com/apache2.0/
+
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+-->
+
+# Support for ShmSize and Tmpfs Docker Parameters in ECS
+
+## Background
+
+Docker supports the parameters `--shm-size` and `--tmpfs` which sets the size of `/dev/shm` volume and creates an empty tmpfs mount within a container respectively. More information on these flags:
+* ShmSize: https://docs.docker.com/engine/reference/run/#runtime-constraints-on-resources
+* Tmpfs: https://docs.docker.com/engine/reference/run/#tmpfs-mount-tmpfs-filesystems
+
+## ECS Support for the flags
+
+### ShmSize
+A new field will be added for Docker's --shm-size flag under the linuxParameters field in ContainerDefinition of the Task definition. Through this field, customers will be able to set the value for `ShmSize` parameter in Docker's HostConfig that is passed during create container call.
+
+### Tmpfs
+A new field will be added for Docker's --tmpfs flag under the linuxParameters field in ContainerDefinition of the Task definition.  Through this field, customers will be able to set the value for `Tmpfs` parameter in Docker's HostConfig that is passed during the create container call.
+
+Tmpfs parameter takes in a container path as the mount point and other mount options for tmpfs including the mount size. Other options can be found in the man page for mount [here](http://man7.org/linux/man-pages/man8/mount.8.html).
+
+## Implications of using these fields during task placement decisions:
+
+Since both the flags consume memory in the instance in which their corresponding tasks are placed, their memory sizes will be accounted against the total memory available in the instance. As a result, the memory set for both the ShmSize and Tmpfs parameters will affect placement decisions for other tasks in the same instances. Size will be a mandatory parameter in the Tmpfs field since it is essential for making task placement decisions.
+
+CloudWatch Metrics would now include additional metrics on memory reserved for these parameters.
```

Comment:
Are you just talking about the reserved memory metric?  Does the utilization metric also get affected by this or does utilization just cover Memory/MemoryReservation fields in the task def?

Reply:
It is just the reservation metrics for both the new parameters.

Reply:
Here's a hypothetical situation: If I have 80 MiB allocated through the "memory" parameter (and I'm using all of it) and I have 20 MiB through ShmSize, what do I see?

I expect that reservation metrics would show 100 MiB (80 + 20).
I'm not sure whether utilization metrics would show 100% (80 / 80) or 80% (80 / 100).

Reply:
In this case, the utilization metrics would still show 100%. If we explicitly specify that the utilization metrics do not include ShmSize and Tmpfs size consumed, it would make sense.